src/v/kafka/server/handlers/create_partitions.cc

```diff
@@ -206,6 +207,29 @@ ss::future<response_ptr> create_partitions_handler::handle(
         co_return co_await ctx.respond(std::move(resp));
     }
 
+    valid_range_end = validate_range(
+      request.data.topics.begin(),
+      valid_range_end,
+      std::back_inserter(resp.data.results),
+      ((ctx.header().version >= api_version(3))
+         ? error_code::throttling_quota_exceeded
+         : error_code::unknown_server_error),
+      "Too many partition mutations requested",
+      [&ctx, &resp](const create_partitions_topic& tp) {
+          const auto cfg = ctx.metadata_cache().get_topic_cfg(
+            model::topic_namespace_view(model::kafka_namespace, tp.name));
+          vassert(cfg, "Topic exist check has already occurred");
+          vassert(
+            tp.count > cfg->partition_count,
+            "Sanity check for request increase partition count failed");
+          const auto mutations = (tp.count - cfg->partition_count);
+          const auto delay = ctx.quota_mgr().record_partition_mutations(
```

Comment:
should this be broker wide enforced rather than shard local? that would probably be the second best case over enforcing rates at the controller itself. when nodes have large numbers of cores the per shard rate limit is hard to work with. per node is easier to reason about.

Reply:
done